app/controllers/EHealthVDCController.scala

```diff
@@ -1,144 +1,202 @@
-      val patientDetailsDF = spark.sql(query)
-      patientDetailsDF.show(false)
-
-      val rawJson = patientDetailsDF.toJSON.collect().mkString
-      Future.successful(Ok(Json.toJson(rawJson)))
+  private def sendRequestToEnforcmentEngine (purpose: String, requesterId: String, url:String, testType: String): String = {
+    var newTestType:String = null
+    if (testType.equals("cholesterol")) {
+      newTestType = "cholesterol_total_value"
+    } else {
+      newTestType = "%s_value".format(testType)
+    }
+    val query = "SELECT patientId, date, %s FROM blood_tests".format(newTestType)
+
+    val data = Json.obj(
+      "query" -> query,
+      "purpose" -> purpose,
+      "access" -> "read",
+      "requester" -> "",
+      "blueprintId" -> "",
+      "requesterId" -> requesterId
+    )
+    val futureResponse = ws.url(url).addHttpHeaders("Content-Type" -> "application/json")
+      .addHttpHeaders("Accept" -> "application/json").withRequestTimeout(Duration.Inf).post(data)
+
+    val res = Await.result(futureResponse, Duration.Inf)
+    res.body[String]
   }
 
   @ApiOperation(nickname = "getAllValuesForBloodTestComponent",
     value = "Get timeseries of patient's blood test component",
     notes =  "This method returns the collected values for a specific blood test component of a patient (identified by his SSN), to be used by medical doctors",
-    response = classOf[models.Patient], responseContainer = "List", httpMethod = "GET")
+    response = classOf[models.BloodTestComponentValue], responseContainer = "List", httpMethod = "GET")
   @ApiResponses(Array(
     new ApiResponse(code = 404, message = "Patient not found")))
-  def getTestValues(@ApiParam(value = "SSN", required = true,
-    allowMultiple = false) socialId: String,
-                    @ApiParam(value = "component", required = true,
-                      allowMultiple = false) testType: String): Action[AnyContent] = {
-    Action.async {
+  def getAllValuesForBloodTestComponent(@ApiParam(value = "SSN", required = true, allowMultiple = false) socialId: String,
+                                        @ApiParam(value = "component", required = true,
+                                          allowMultiple = false) testType: String)= Action.async {
+    implicit request =>
       val spark = initService.getSparkSessionInstance
-      readData(spark)
-      val query = "select patientId, date, %s.value as %s from joined where socialId='%s'".format(testType, testType, socialId)
-      val patientBloodTestsDF = spark.sql(query)
-      patientBloodTestsDF.limit(10).show(false)
-      patientBloodTestsDF.printSchema
-       patientBloodTestsDF.explain(true)
-
-      val rawJson = patientBloodTestsDF.limit(10).toJSON.collect().mkString
-      Future.successful(Ok(Json.toJson(rawJson)))
-    }
+      val patientSSN = socialId
+      var origtestType = testType
+
+      debugMode = initService.getDebugMode
+      if (config.has("policy.enforcement.play.url")) {
```

Comment:
debugMode and enforcementEngineURL should be class fields initialized in constructor, not in every method.